libbeat/autodiscover/providers/kubernetes/config.go

```diff
@@ -30,11 +30,16 @@ import (
 // Config for kubernetes autodiscover provider
 type Config struct {
 	KubeConfig     string        `config:"kube_config"`
-	Host           string        `config:"host"`
 	Namespace      string        `config:"namespace"`
 	SyncPeriod     time.Duration `config:"sync_period"`
 	CleanupTimeout time.Duration `config:"cleanup_timeout" validate:"positive"`
 
+	// Needed when resource is a pod
+	Host string `config:"host"`
+	// Scope can be either host or cluster.
+	Scope    string `config:"scope"`
```

Comment:
This would also solve https://github.com/elastic/beats/issues/10578 :tada:

Reply:
can happen in a different PR but I think we should extend this setting to `add_kubernetes_metadata` processor. I'm unsure about the other (`resource`), probably not worth it until we see the need.

Reply:
will create follow up PR